In [ ]:
!pip install -q pyomo

In [ ]:
from pyomo.environ import *

In [ ]:
# create a model
model = ConcreteModel()

In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2)
model.x2 = Var(initialize=2)
model.x3 = Var(initialize=2)
model.x4 = Var(initialize=2)
model.x5 = Var(initialize=2)

In [ ]:
model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

In [ ]:
def funct(model):
  return (model.x1-1.0)**2 + (model.x1 - model.x2)**2 + (model.x2 - model.x3)**2 + (model.x3 - model.x4)**4 + (model.x4 - model.x5)**4

In [ ]:
# declare objective
model.cost = Objective(rule = funct, sense=minimize)

In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     2 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     2 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     2 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     2 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     2 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
#We will use ipopt solver for solving the optimization problem

!wget -N -q "https://ampl.com/dl/open/ipopt/ipopt-linux64.zip"
!unzip -o -q ipopt-linux64

In [ ]:
solver = SolverFactory('ipopt', executable='/content/ipopt')

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.028654098510742188
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())

print("Constraints:")
print("Constraint 1 is active because LHS evaluates to",model.constr1(),"= RHS = 0")
print("Constraint 2 is active because LHS evaluates to",model.constr2(),"= RHS = 0")
print("Constraint 3 is active because LHS evaluates to",model.constr3(),"= RHS = 0")
print("Constraint 4 is inactive because LHS evaluates to",model.constr4(),"> RHS = -5")
print("Constraint 5 is inactive because LHS evaluates to",model.constr5(),"< RHS = 5")
print("Constraint 6 is inactive because LHS evaluates to",model.constr6(),"> RHS = 0")
print("Constraint 7 is inactive because LHS evaluates to",model.constr7(),"< RHS = 3")


Cost =  0.07877682087105699

Optimal solution of the function:
x1 =  1.1911274562403504
x2 =  1.3626031649434767
x3 =  1.4728179315306016
x4 =  1.6350166192407933
x5 =  1.6790814362660684
Constraints:
Constraint 1 is active because LHS evaluates to 1.7763568394002505e-15 = RHS = 0
Constraint 2 is active because LHS evaluates to -4.440892098500626e-16 = RHS = 0
Constraint 3 is active because LHS evaluates to -3.774758283725532e-15 = RHS = 0
Constraint 4 is inactive because LHS evaluates to 1.1911274562403504 > RHS = -5
Constraint 5 is inactive because LHS evaluates to 1.1911274562403504 < RHS = 5
Constraint 6 is inactive because LHS evaluates to 1.4728179315306016 > RHS = 0
Constraint 7 is inactive because LHS evaluates to 1.4728179315306016 < RHS = 3


In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2**2)
model.x2 = Var(initialize=2**2)
model.x3 = Var(initialize=2**2)
model.x4 = Var(initialize=2**2)
model.x5 = Var(initialize=2**2)

model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

# declare objective
model.cost = Objective(rule = funct, sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     4 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     4 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     4 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     4 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :     4 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.03763914108276367
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())


Cost =  0.07877682087105696

Optimal solution of the function:
x1 =  1.1911274562403504
x2 =  1.3626031649434764
x3 =  1.4728179315306016
x4 =  1.6350166192407931
x5 =  1.6790814362660684


In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2**4)
model.x2 = Var(initialize=2**4)
model.x3 = Var(initialize=2**4)
model.x4 = Var(initialize=2**4)
model.x5 = Var(initialize=2**4)

model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

# declare objective
model.cost = Objective(rule = funct, sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :    16 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :    16 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :    16 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :    16 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :    16 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.08252239227294922
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())


Cost =  0.07877682087105699

Optimal solution of the function:
x1 =  1.1911274562403504
x2 =  1.3626031649434767
x3 =  1.4728179315306016
x4 =  1.6350166192407933
x5 =  1.6790814362660684


In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2**8)
model.x2 = Var(initialize=2**8)
model.x3 = Var(initialize=2**8)
model.x4 = Var(initialize=2**8)
model.x5 = Var(initialize=2**8)

model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

# declare objective
model.cost = Objective(rule = funct, sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   256 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   256 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   256 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   256 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :   256 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.08246326446533203
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())


Cost =  0.07877682087275217

Optimal solution of the function:
x1 =  1.1911274559325666
x2 =  1.3626031648437267
x3 =  1.472817931626912
x4 =  1.6350166196164382
x5 =  1.6790814366888756


In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2**10)
model.x2 = Var(initialize=2**10)
model.x3 = Var(initialize=2**10)
model.x4 = Var(initialize=2**10)
model.x5 = Var(initialize=2**10)

model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

# declare objective
model.cost = Objective(rule = funct, sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  1024 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  1024 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  1024 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  1024 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  1024 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.0760340690612793
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())


Cost =  0.07877682087107182

Optimal solution of the function:
x1 =  1.191127454862753
x2 =  1.3626031645868748
x3 =  1.4728179318916943
x4 =  1.6350166206609638
x5 =  1.6790814382066548


In [ ]:
# declare decision variables with initialization
model.x1 = Var(initialize=2**12)
model.x2 = Var(initialize=2**12)
model.x3 = Var(initialize=2**12)
model.x4 = Var(initialize=2**12)
model.x5 = Var(initialize=2**12)

model.constr1 = Constraint(expr=model.x1 + model.x2**2 + model.x3**3 - 2 - 3*((2)**(1/2))== 0)
model.constr2 = Constraint(expr=model.x2 - model.x3**2 + model.x4 + 2 - 2*((2)**(1/2))== 0)
model.constr3 = Constraint(expr=model.x1*model.x5 - 2 == 0)
model.constr4 = Constraint(expr=model.x1 >= -5)
model.constr5 = Constraint(expr=model.x1 <= 5)
model.constr6 = Constraint(expr=model.x3 >= 0)
model.constr7 = Constraint(expr=model.x3 <= 3)

# declare objective
model.cost = Objective(rule = funct, sense=minimize)

This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().
This is usually indicative of a modelling error.
To avoid this warning, use block.del_component() and block.add_component().


In [ ]:
model.pprint()

5 Var Declarations
    x1 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  4096 :  None : False : False :  Reals
    x2 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  4096 :  None : False : False :  Reals
    x3 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  4096 :  None : False : False :  Reals
    x4 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  4096 :  None : False : False :  Reals
    x5 : Size=1, Index=None
        Key  : Lower : Value : Upper : Fixed : Stale : Domain
        None :  None :  4096 :  None : False : False :  Reals

1 Objective Declarations
    cost : Size=1, Index=None, Active=True
        Key  : Active : Sense    : Expression
        None :   True : minimize : (x1 - 1.0)**2 + (x1 - x2)**2 + (x2 - x3)**2 + (x3 - x4)**4 + (x4 - x5)*

In [ ]:
results = solver.solve(model)

In [ ]:
#display the statistics 
print(results)


Problem: 
- Lower bound: -inf
  Upper bound: inf
  Number of objectives: 1
  Number of constraints: 7
  Number of variables: 5
  Sense: unknown
Solver: 
- Status: ok
  Message: Ipopt 3.12.13\x3a Optimal Solution Found
  Termination condition: optimal
  Id: 0
  Error rc: 0
  Time: 0.13318204879760742
Solution: 
- number of solutions: 0
  number of solutions displayed: 0



In [ ]:
# display solution
print('\nCost = ', model.cost())

print('\nOptimal solution of the function:')
print('x1 = ', model.x1())
print('x2 = ', model.x2())
print('x3 = ', model.x3())
print('x4 = ', model.x4())
print('x5 = ', model.x5())


Cost =  0.078776821476022

Optimal solution of the function:
x1 =  1.1911275027451427
x2 =  1.3626031771640859
x3 =  1.4728179218935038
x4 =  1.6350165752432628
x5 =  1.6790813649752658


Answer 3)

Status: ok

Termination condition: optimal

Time: 0.03863859176635742

Answer 4)

Cost =  0.07877682087105699

Optimal solution of the function:

x1 =  1.1911274562403504

x2 =  1.3626031649434767

x3 =  1.4728179315306016

x4 =  1.6350166192407933

x5 =  1.6790814362660684



Answer 5)

Constraint 1 is active because LHS evaluates to 1.7763568394002505e-15 = RHS = 0

Constraint 2 is active because LHS evaluates to -4.440892098500626e-16 = RHS = 0

Constraint 3 is active because LHS evaluates to -3.774758283725532e-15 = RHS = 0

Constraint 4 is inactive because LHS evaluates to 1.1911274562403504 > RHS = -5

Constraint 5 is inactive because LHS evaluates to 1.1911274562403504 < RHS = 5

Constraint 6 is inactive because LHS evaluates to 1.4728179315306016 > RHS = 0

Constraint 7 is inactive because LHS evaluates to 1.4728179315306016 < RHS = 3

Answer 6)

i=2

Cost =  0.07877682087105696

Optimal solution of the function:

x1 =  1.1911274562403504

x2 =  1.3626031649434764

x3 =  1.4728179315306016

x4 =  1.6350166192407931

x5 =  1.6790814362660684

Time: 0.03763914108276367

i=4

Cost =  0.07877682087105699

Optimal solution of the function:

x1 =  1.1911274562403504

x2 =  1.3626031649434767

x3 =  1.4728179315306016

x4 =  1.6350166192407933

x5 =  1.6790814362660684

Time: 0.08252239227294922

i=8

Cost =  0.07877682087275217

Optimal solution of the function:

x1 =  1.1911274559325666

x2 =  1.3626031648437267

x3 =  1.472817931626912

x4 =  1.6350166196164382

x5 =  1.6790814366888756

Time: 0.08246326446533203


i=10


Cost =  0.07877682087107182

Optimal solution of the function:

x1 =  1.191127454862753

x2 =  1.3626031645868748

x3 =  1.4728179318916943

x4 =  1.6350166206609638

x5 =  1.6790814382066548

Time: 0.0760340690612793

i=12

Cost =  0.078776821476022

Optimal solution of the function:

x1 =  1.1911275027451427

x2 =  1.3626031771640859

x3 =  1.4728179218935038

x4 =  1.6350165752432628

x5 =  1.6790813649752658

Time: 0.13318204879760742

Observation: Time taken is more when distance between optimal point and initial point is more (exception is last case when i=12).
